### PR TITLE
add elementsOf to contains.ignoreCase fluent-api

### PR DESCRIPTION
**Pull request by Miftahunajat**
Following the things that I done :

_fluent-api_

- [x] add elementsOf to charSequenceContainsCreators for CharSequenceContains.Builder<T, IgnoringCaseSearchBehaviour> (copy the existing elementsOf and adjust)
- [x] search for TODO #422 in specs and fix
- [x] add test-case to CharSequenceContainsAtLeastAssertionsSpec covering the case that an empty iterable is passed

infix-api
- [ ] add elementsOf to charSequenceContainsCreators for CharSequenceContains.Builder<T, IgnoringCaseSearchBehaviour> (copy the existing elementsOf and adjust)
- [ ] search for TODO #422 in specs and fix
- [ ] add test-case to CharSequenceContainsAtLeastAssertionsSpec covering the case that an empty iterable is passed
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

**Comment by robstoll** [comment]
> add test-case to CharSequenceContainsAtLeastAssertionsSpec covering the case that an empty iterable is passed

This one is not yet done, you can copy an existing one, e.g. `describe("ignoringCase.atLeast(1).elementsOf")`

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsAtLeastAssertionsSpec.kt [hunk 1]
```diff
@@ -101,7 +101,12 @@ class CharSequenceContainsAtLeastAssertionsSpec : Spek({
             a: Any,
             aX: Array<out Any>
         ): Expect<CharSequence> =
-            expect.contains.ignoringCase.atLeast(atLeast).elementsOf(listOf(a, *aX))
+            if (aX.isEmpty()) {
+                expect.contains.ignoringCase.atLeast(atLeast).elementsOf(listOf(a))
```
[the remark this answers is not in the document]
Comment (Miftahunajat) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsAtLeastAssertionsSpec.kt:105]: Sorry, can you please elaborate more about the shortcut?

**Comment by Miftahunajat** [comment]
> > add test-case to CharSequenceContainsAtLeastAssertionsSpec covering the case that an empty iterable is passed
> 
> This one is not yet done, you can copy an existing one, e.g. `describe("ignoringCase.atLeast(1).elementsOf")`

after I copy the previous one, should I change it from `                    expect("test").contains.ignoringCase.atLeast(1).elementsOf(emptyList())
` to 
`                    
expect("test").contains.ignoringCase.elementsOf(emptyList())
`?
to cover the case?

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsAtLeastAssertionsSpec.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (robstoll) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsAtLeastAssertionsSpec.kt:105]: copy line 87 etc. and adopt to ignoringCase, if atLeast == 1 then we want to leave out `atLeast(atLeast)` (as in line 87)

**Comment by robstoll** [comment]
> after I copy the previous one, should I change it from ` expect("test").contains.ignoringCase.atLeast(1).elementsOf(emptyList())` to
> ` expect("test").contains.ignoringCase.elementsOf(emptyList())`?
> to cover the case?

Exactly, with shortcut I meant leaving out `atLeast(1)` in this context. This should also answer your first question

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsAtLeastAssertionsSpec.kt [hunk 2]
```diff
@@ -36,10 +36,10 @@ class CharSequenceContainsAtLeastAssertionsSpec : Spek({
                 }.toThrow<IllegalArgumentException> { messageContains("Iterable without elements are not allowed") }
             }
         }
-        describe("ignoringCase.atLeast(1).elementsOf") {
+        describe("ignoringCase.elementsOf") {
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsAtLeastAssertionsSpec.kt:39]: a misunderstanding I guess. Please don't adjust the existing test but copy it and adjust it then.

Comment (Miftahunajat) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsAtLeastAssertionsSpec.kt:39]: sorry it was my mistake, I thought I had already copied it

Comment (robstoll) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsAtLeastAssertionsSpec.kt:46]: no worries

**Comment by robstoll** [comment]
@Miftahunajat do you want to do the infix-api in this PR? if not, then please remove the WIP prefix in your title and I am going to merge it.

**Comment by Miftahunajat** [comment]
> @Miftahunajat do you want to do the infix-api in this PR? if not, then please remove the WIP prefix in your title and I am going to merge it.

Okay, I will remove it. I will do the infix-api on another PR.
Thank you

**Comment by robstoll** [comment]
@Miftahunajat thanks for your first contribution :tada:
